src/Data/Attoparsec/Args.hs

```diff
@@ -48,15 +97,41 @@ argsParser mode = many (P.skipSpace *> (quoted <|> unquoted)) <*
     nonquote = P.satisfy (not . (=='"'))
     naked = P.satisfy (not . flip elem ("\" " :: String))
 
+-- | Parser to extract the stack command line embedded inside a comment
+-- after validating the placement and formatting rules for a valid
+-- interpreter specification.
+
+interpParser :: String -> Parser Text String
```

Comment:
Minor nitpick: How about `interpreterArgsParser`?  While we do abbreviate things here and there, it's a bit arbitrary to shorten interpreter to `interp`.